#VRML_SIM R2023a utf8

EXTERNPROTO "../protos/TexturedBackgroundLight.proto"
EXTERNPROTO "../protos/TexturedBackground.proto"
EXTERNPROTO "../protos/halfTile.proto"
EXTERNPROTO "../protos/HazardMap.proto"
EXTERNPROTO "../protos/Victim.proto"
EXTERNPROTO "../protos/worldTile.proto"
IMPORTABLE EXTERNPROTO "../protos/custom_robot.proto"
EXTERNPROTO "../protos/E-puck.proto"

WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "TILE"
      material2 "NO_FRIC"
      coulombFriction [
        0
      ]
      bounce 0
      bumpSound ""
      rollSound ""
      slideSound ""
    }
  ]
}
DEF Viewpoint Viewpoint {
  orientation -0.42705998397599537 0.6732426844501265 0.6036257598221021 3.735065671078827
  position 0.21403013809281 0.6218096775760824 0.4217173424617024
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF WALLTILES Group {
  children [
    DEF START_TILE worldTile {
      floor TRUE
      topWall 1
      leftWall 1
      topExternal TRUE
      leftExternal TRUE
      start TRUE
      width 8
      height 8
      id "0"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      zPos 1
      floor TRUE
      leftWall 1
      leftExternal TRUE
      width 8
      height 8
      id "1"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      zPos 2
      floor TRUE
      leftWall 1
      leftExternal TRUE
      width 8
      height 8
      id "2"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      zPos 3
      floor TRUE
      leftWall 1
      leftExternal TRUE
      width 8
      height 8
      id "3"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      zPos 4
      floor TRUE
      leftWall 1
      leftExternal TRUE
      width 8
      height 8
      id "4"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      zPos 5
      floor TRUE
      leftWall 1
      leftExternal TRUE
      width 8
      height 8
      id "5"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      zPos 6
      floor TRUE
      leftWall 1
      leftExternal TRUE
      width 8
      height 8
      id "6"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      zPos 7
      floor TRUE
      bottomWall 5
      leftWall 3
      leftExternal TRUE
      bottomExternal TRUE
      width 8
      height 8
      id "7"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 1
      floor TRUE
      topWall 1
      topExternal TRUE
      width 8
      height 8
      id "8"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 1
      zPos 1
      floor TRUE
      rightWall 3
      bottomWall 3
      width 8
      height 8
      id "9"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 1
      zPos 2
      floor TRUE
      rightWall 2
      width 8
      height 8
      id "10"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 1
      zPos 3
      floor TRUE
      rightWall 1
      width 8
      height 8
      id "11"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 1
      zPos 4
      floor TRUE
      width 8
      height 8
      id "12"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 1
      zPos 5
      floor TRUE
      rightWall 3
      width 8
      height 8
      id "13"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 1
      zPos 6
      floor TRUE
      width 8
      height 8
      id "14"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 1
      zPos 7
      floor TRUE
      bottomWall 1
      bottomExternal TRUE
      width 8
      height 8
      id "15"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 2
      floor TRUE
      topWall 1
      topExternal TRUE
      width 8
      height 8
      id "16"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 2
      zPos 1
      floor TRUE
      width 8
      height 8
      id "17"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 2
      zPos 2
      floor TRUE
      width 8
      height 8
      id "18"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 2
      zPos 3
      floor TRUE
      width 8
      height 8
      id "19"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 2
      zPos 4
      floor TRUE
      width 8
      height 8
      id "20"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 2
      zPos 5
      floor TRUE
      rightWall 3
      bottomWall 15
      width 8
      height 8
      id "21"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 2
      zPos 6
      floor TRUE
      rightWall 2
      width 8
      height 8
      id "22"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 2
      zPos 7
      floor TRUE
      bottomWall 1
      bottomExternal TRUE
      width 8
      height 8
      id "23"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 3
      rightWall 2
      topWall 3
      topExternal TRUE
      trap TRUE
      width 8
      height 8
      id "24"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 3
      zPos 1
      floor TRUE
      rightWall 3
      bottomWall 3
      width 8
      height 8
      id "25"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 3
      zPos 2
      floor TRUE
      bottomWall 3
      width 8
      height 8
      id "26"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.10196078431372549 0.10196078431372549 0.9019607843137255
      room 1
    }
    DEF TILE worldTile {
      xPos 3
      zPos 3
      floor TRUE
      rightWall 6
      width 8
      height 8
      id "27"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 3
      zPos 4
      floor TRUE
      width 8
      height 8
      id "28"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 3
      zPos 5
      floor TRUE
      width 8
      height 8
      id "29"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 3
      zPos 6
      floor TRUE
      width 8
      height 8
      id "30"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 3
      zPos 7
      floor TRUE
      bottomWall 1
      bottomExternal TRUE
      width 8
      height 8
      id "31"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE halfTile {
      xPos 4
      id "32"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      topWall 6
      tile2Walls [
        0, 2, 0, 0
      ]
      tile1External [
        TRUE
        FALSE
        FALSE
        FALSE
      ]
      tile2External [
        TRUE
        FALSE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 4
      zPos 1
      id "33"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 4
      zPos 2
      id "34"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tile3Walls [
        0, 0, 2, 0
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 4
      zPos 3
      id "35"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      bottomWall 15
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE worldTile {
      xPos 4
      zPos 4
      floor TRUE
      rightWall 2
      width 8
      height 8
      id "36"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 4
      zPos 5
      floor TRUE
      rightWall 1
      width 8
      height 8
      id "37"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 4
      zPos 6
      floor TRUE
      rightWall 1
      width 8
      height 8
      id "38"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE worldTile {
      xPos 4
      zPos 7
      floor TRUE
      rightWall 3
      bottomWall 3
      bottomExternal TRUE
      checkpoint TRUE
      width 8
      height 8
      id "39"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.635 0.635 0.635
      room 1
    }
    DEF TILE halfTile {
      xPos 5
      id "40"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      topWall 2
      tile1External [
        TRUE
        FALSE
        FALSE
        FALSE
      ]
      tile2External [
        TRUE
        FALSE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 5
      zPos 1
      id "41"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tile4Walls [
        0, 3, 0, 0
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 5
      zPos 2
      id "42"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tile2Walls [
        0, 2, 0, 0
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 5
      zPos 3
      id "43"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      rightWall 3
      bottomWall 6
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 5
      zPos 4
      id "44"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 5
      zPos 5
      id "45"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 5
      zPos 6
      id "46"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 5
      zPos 7
      id "47"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      bottomWall 2
      tile3External [
        FALSE
        FALSE
        TRUE
        FALSE
      ]
      tile4External [
        FALSE
        FALSE
        TRUE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 6
      id "48"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      topWall 1
      tile1External [
        TRUE
        FALSE
        FALSE
        FALSE
      ]
      tile2External [
        TRUE
        FALSE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 6
      zPos 1
      id "49"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tile3Walls [
        0, 0, 5, 0
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 6
      zPos 2
      id "50"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE worldTile {
      xPos 6
      zPos 3
      floor TRUE
      rightWall 3
      width 8
      height 8
      id "51"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tileColor 0.30196078431372547 0.10196078431372549 0.6
      room 1
    }
    DEF TILE halfTile {
      xPos 6
      zPos 4
      id "52"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 6
      zPos 5
      id "53"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      tile1Walls [
        0, 3, 0, 0
      ]
      tile2Walls [
        0, 0, 5, 0
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 6
      zPos 6
      id "54"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      width 8
      height 8
      floor TRUE
      curve [
        0, 2, 4, 0
      ]
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 6
      zPos 7
      id "55"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      bottomWall 1
      tile3External [
        FALSE
        FALSE
        TRUE
        FALSE
      ]
      tile4External [
        FALSE
        FALSE
        TRUE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 7
      id "56"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      rightWall 2
      topWall 3
      tile1External [
        TRUE
        FALSE
        FALSE
        FALSE
      ]
      tile2External [
        TRUE
        TRUE
        FALSE
        FALSE
      ]
      tile4External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      swamp TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 7
      zPos 1
      id "57"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      rightWall 1
      tile2External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      tile4External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 7
      zPos 2
      id "58"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      rightWall 1
      tile2External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      tile4External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 7
      zPos 3
      id "59"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      rightWall 3
      bottomWall 15
      tile2External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      tile4External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 2
    }
    DEF TILE halfTile {
      xPos 7
      zPos 4
      id "60"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      rightWall 2
      tile2External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      tile4External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 7
      zPos 5
      id "61"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      rightWall 1
      tile2External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      tile4External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 7
      zPos 6
      id "62"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      rightWall 1
      tile2External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      tile4External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
    DEF TILE halfTile {
      xPos 7
      zPos 7
      id "63"
      xScale 0.4
      zScale 0.4
      yScale 0.4
      rightWall 3
      bottomWall 3
      tile2External [
        FALSE
        TRUE
        FALSE
        FALSE
      ]
      tile3External [
        FALSE
        FALSE
        TRUE
        FALSE
      ]
      tile4External [
        FALSE
        TRUE
        TRUE
        FALSE
      ]
      width 8
      height 8
      floor TRUE
      tileColor 0.635 0.635 0.635
      room 3
    }
  ]
}
DEF CHECKPOINTBOUNDS Group {
  children [
    DEF boundary Group {
      children [
        DEF checkpoint0min Transform {
          translation -0.06 -0.03 0.3000000000000001
        }
        DEF checkpoint0max Transform {
          translation 0.06 -0.03 0.4200000000000001
        }
      ]
    }
  ]
}
DEF TRAPBOUNDS Group {
  children [
    DEF boundary Group {
      children [
        DEF trap0min Transform {
          translation -0.18 -0.03 -0.54
        }
        DEF trap0max Transform {
          translation -0.06 -0.03 -0.42
        }
      ]
    }
  ]
}
DEF STARTBOUNDS Group {
  children [
    DEF boundary Group {
      children [
        DEF start0min Transform {
          translation -0.54 -0.03 -0.54
        }
        DEF start0max Transform {
          translation -0.42 -0.03 -0.42
        }
      ]
    }
  ]
}
DEF SWAMPBOUNDS Group {
  children [
    DEF boundary Group {
      children [
        DEF swamp0min Transform {
          translation 0.3000000000000001 -0.03 -0.54
        }
        DEF swamp0max Transform {
          translation 0.4200000000000001 -0.03 -0.42
        }
      ]
    }
  ]
}
DEF OBSTACLES Group {
}
DEF HUMANGROUP Group {
  children [
    Victim {
      translation -0.23399999999999999 0 0.1750000000000001
      name "Victim0"
      scoreWorth 15
    }
    Victim {
      translation 0.0650000000000001 0 0.13300000000000012
      rotation 0 1 0 -1.57
      name "Victim1"
      type "unharmed"
      scoreWorth 5
    }
    Victim {
      translation 0.4150000000000001 0 -0.372
      rotation 0 1 0 1.57
      name "Victim2"
      type "stable"
      scoreWorth 5
    }
  ]
}
DEF HAZARDGROUP Group {
  children [
    HazardMap {
      translation -0.5344 0 -0.258
      rotation 0 1 0 -1.57
      name "Hazard0"
    }
    HazardMap {
      translation -0.5344 0 0.262
      rotation 0 1 0 -1.57
      name "Hazard1"
      type "P"
    }
  ]
}
DEF MAINSUPERVISOR Robot {
  children [
    Receiver {
      channel 1
    }
    Emitter {
      channel 1
    }
  ]
  controller "MainSupervisor"
  customData "480"
  supervisor TRUE
  window "MainSupervisorWindow"
}
E-puck {
  hidden position_0_0 5.95173783354352
  hidden position_0_1 -3.767162375150353
  hidden linearVelocity_0 0.00029261050787693816 1.3236154927701229e-05 4.7307889724319e-05
  hidden angularVelocity_0 0.0031538530000910616 -1.1818063475628076e-06 -0.019507338479112263
  hidden translation_1 0 0.02599999999999971 0.02
  hidden rotation_1 0 -1 0 1.502870148635807
  hidden linearVelocity_1 0.00039014540268470004 -9.432365683448307e-06 6.30870376330371e-05
  hidden angularVelocity_1 0.0031538363311427718 -1.1819999609153404e-06 -0.019507340393119563
  hidden translation_2 0 -0.02599999999999981 0.02
  hidden rotation_2 0 1 0 4.96133556327404
  hidden linearVelocity_2 0.00039014780300490853 3.723809839106806e-05 6.306721701102234e-05
  hidden angularVelocity_2 0.0031538696797323404 -1.1816127068546844e-06 -0.01950733655060078
  translation -0.09683091335583027 -0.030032830263611345 0.3605851966751113
  rotation -0.04066169525269098 0.7065448694704788 0.7064990969308603 3.0604897889678653
  controller "lidarTest"
  camera_fieldOfView 0.9
  camera_width 1280
  camera_height 720
  turretSlot [
    InertialUnit {
      name "iunit"
    }
    DistanceSensor {
      rotation 0 0 1 -1.5708
      name "laser1"
      lookupTable [
        0 0 0
        0.1 0.1 0
        0.2 0.2 0
        1 1 0
      ]
      type "laser"
      aperture 0.01
    }
    DistanceSensor {
      rotation 0 0 1 -1.0472
      name "laser2"
      lookupTable [
        0 0 0
        0.1 0.1 0
        0.2 0.2 0
        1 1 0
      ]
      type "laser"
      aperture 0.01
    }
    DistanceSensor {
      rotation 0 0 1 -0.523599
      name "laser3"
      lookupTable [
        0 0 0
        0.1 0.1 0
        0.2 0.2 0
        1 1 0
      ]
      type "laser"
      aperture 0.01
    }
    DistanceSensor {
      rotation 0 0 1 -0.261799
      name "laser4"
      lookupTable [
        0 0 0
        0.1 0.1 0
        0.2 0.2 0
        1 1 0
      ]
      type "laser"
      aperture 0.01
    }
    DistanceSensor {
      rotation 0 0 1 0.261799
      name "laser5"
      lookupTable [
        0 0 0
        0.1 0.1 0
        0.2 0.2 0
        1 1 0
      ]
      type "laser"
      aperture 0.01
    }
    DistanceSensor {
      rotation 0 0 1 0.523599
      name "laser6"
      lookupTable [
        0 0 0
        0.1 0.1 0
        0.2 0.2 0
        1 1 0
      ]
      type "laser"
      aperture 0.01
    }
    DistanceSensor {
      rotation 0 0 1 1.0472
      name "laser7"
      lookupTable [
        0 0 0
        0.1 0.1 0
        0.2 0.2 0
        1 1 0
      ]
      type "laser"
      aperture 0.01
    }
    DistanceSensor {
      rotation 0 0 1 1.5708
      name "laser8"
      lookupTable [
        0 0 0
        0.1 0.1 0
        0.2 0.2 0
        1 1 0
      ]
      type "laser"
      aperture 0.01
    }
    DistanceSensor {
      name "laser9"
      lookupTable [
        0 0 0
        0.1 0.1 0
        0.2 0.2 0
        1 1 0
      ]
      type "laser"
      aperture 0.01
    }
    Compass {
    }
    Display {
      width 1000
      height 1000
    }
    LightSensor {
      translation 0.03 0.03 0
      name "left_heat_sensor"
      lookupTable [
        37 12 0
        30 5 0
        25 2 0
        21 0 0
        38 1000 0
      ]
    }
    LightSensor {
      translation 0.03 -0.03 0
      name "right_heat_sensor"
      lookupTable [
        37 12 0
        30 5 0
        25 2 0
        21 0 0
        38 1000 0
      ]
    }
    Camera {
      translation 0.04 0 -0.03
      rotation 0 1 0 0.52
      name "colour_camera"
      recognitionColors [
        0.713726 0.580392 0.32549
        0.505882 0 0.760784
        1 0 0
        0 1 0
      ]
      recognition Recognition {
      }
    }
    GPS {
    }
    Camera {
      translation 0.0336617 0.0163406 -0.022
      rotation 0 0 1 0.785405
      name "camera_left"
      recognitionColors [
        1 1 1
      ]
      width 1280
      height 720
      recognition Recognition {
      }
    }
    Camera {
      translation 0.0325533 -0.0203553 -0.022
      rotation 0 0 -1 0.785396
      name "camera_right"
      recognitionColors [
        1 1 1
      ]
      width 1280
      height 720
      recognition Recognition {
      }
    }
  ]
}
